web/modules/custom/grundsalg_db_client/src/Controller/ApiController.php

```diff
@@ -45,21 +85,122 @@ public function subdivisionUpdated($sid) {
       }
 
       // Create the content.
-      $contentCreationService = \Drupal::service('grundsalg.content_creation');
-      $operation = $contentCreationService->updateSubdivision((array) $content);
+      $operation = $this->contentCreationService->updateSubdivision((array)$content);
 
       return new JsonResponse([
-        'error' => FALSE,
+        'error' => false,
         'message' => $operation,
         'body' => $body,
       ]);
+    } catch (\Throwable $throwable) {
+      return $this->error($throwable);
     }
-    catch (\Exception $e) {
-      return new JsonResponse([
-        'error' => TRUE,
-        'message' => $e->getMessage(),
-        'body' => $e->getMessage(),
-      ]);
+  }
+
+  public function grunde($sid)
+  {
+    try {
+      $request = $this->requestStack->getCurrentRequest();
+
+      $path = sprintf('%s/grunde.json', $sid);
+      if ('POST' === $request->getMethod()) {
+        $this->checkAuthorization();
+        $content = $this->writeFile($path, $request->getContent());
+        return new JsonResponse($content);
```

Comment:
Are you sure this will return correct json? https://symfony.com/doc/current/components/http_foundation.html#creating-a-json-response

`json_encode()` will be called on `$content` https://github.com/symfony/symfony/blob/d3488aff4f34d563c6914583b8e274fccc057c33/src/Symfony/Component/HttpFoundation/JsonResponse.php#L128

Reply:
Nice catch. Fixed in https://github.com/grundsalg-aarhus/grundsalg_web/pull/13/commits/01620aaa9af16a40ea886a46537d85775bb032ad.